test/test_prototype_transforms_functional.py

```diff
@@ -583,6 +584,4 @@ def test_correctness_rotate_bounding_box_on_fixed_input(device, expand):
         center=center,
     )
 
-    assert len(output_boxes) == len(expected_bboxes)
-    for a_out_box, out_box in zip(expected_bboxes, output_boxes.cpu()):
-        np.testing.assert_allclose(out_box.cpu().numpy(), a_out_box)
+    torch.testing.assert_close(output_boxes.tolist(), expected_bboxes)
```

Comment:
`assert_close` refuses to compare a list to a tensor?
IIRC numpy testing utils allow to compare array-likes with array-likes right?

Reply:
Yes and yes. `numpy` testing functions coerce everything into an `np.ndarray` first and than compare. We consciously opted against that for mainly two reasons:

1. If you have tests parametrized over operators like PyTorch core does, it is impossible to see if the returned type is correct. For example

```py
>>> expected = 1.0
>>> np.testing.assert_allclose(np.array(expected), expected)
>>> torch.testing.assert_close(torch.tensor(expected), expected)
TypeError: No comparison pair was able to handle inputs of type <class 'torch.Tensor'> and <class 'float'>.
```

2. It allows us to compare sequences of tensors with different shapes elementwise

    ```py
    >>> shapes = [(2, 3), (2, 4)]
    >>> np.testing.assert_allclose(*([np.ones(shape) for shape in shapes],) * 2)
    ValueError: could not broadcast input array from shape (2,3) into shape (2,)
    >>> torch.testing.assert_close(*([torch.ones(shape) for shape in shapes],) * 2)
    ```

     This aligns nicely with the ability of `torch.testing.assert_close` to compare mappings elementwise, which is not possible with the `numpy` functions.